docs/PadroesDeProjeto/Criacionais/3.1.4Prototype.md

```diff
@@ -0,0 +1,111 @@
+
+Sendo assim, este documento tem como finalidade **explicar o conceito do padrão Prototype, descrever suas principais características, ilustrar suas aplicações práticas e apresentar diagramas UML** que facilitam a compreensão e a implementação deste padrão.  
+
+## Metodologia
+Para a elaboração deste documento, foram seguidas as seguintes etapas:
+
+**Início do Projeto:** Primeiramente, foi realizado um estudo nos slides postados na plataforma Aprender, com o material disponibilizado pela professora <a href="#ref3"><sup>[3]</sup></a>, como base para o entendimento do conceito do padrão de projeto Prototype.
+
+**Busca por Ferramenta:** Após a conclusão dos estudos iniciais, iniciou-se a busca por uma plataforma adequada para ilustrar o padrão Prototype na forma de diagrama UML, com a finalidade de melhor representar o conceito de maneira visual e clara. A plataforma escolhida para isso foi o Lucidchart.
+
+**Análise do Diagrama de Classe:** Em seguida, foi realizada uma análise no diagrama de classes do projeto. Durante a análise, foi observado que não havia classes que pudessem ser aplicadas ao conceito do padrão Prototype <a href="#ref5"><sup>[5]</sup></a>. Esse ponto foi crucial para a decisão de criar um artefato explicativo.
+
+**Desenvolvimento do Artefato:** Como resultado, foi desenvolvido um artefato explicativo utilizando o Lucidchart, com o objetivo de demonstrar o funcionamento do padrão Prototype e suas aplicabilidades, dado que o diagrama de classes do projeto não contemplava diretamente o uso deste padrão.
+
+
+
+## Desenvolvimento
+
+
+<font size="2"><p style="text-align: center">Figura 1 - imagem.</p></font>
+
+<center>
+
+![Prototype](assents/Prototype.svg)
+
+</center>
+
+<font size="2"><p style="text-align: center">Autor(es): [Pedro Paulo](PedroPGH), 2024.</p></font>
+
+### Código do Prototype
+```ts
+export interface Prototype {
+  clone(): Prototype;
+}
+```
+### Código da ConcretePrototypeA
+```ts
+import { Prototype } from './prototype.interface';
+
+export class ConcretePrototypeA implements Prototype {
+    public stateA: string;
+
+    constructor(stateA: string) {
+        this.stateA = stateA;
+    }
+
+    clone(): ConcretePrototypeA {
+        return new ConcretePrototypeA(this.stateA);
+    }
+}
+```
+### Código da ConcretePrototypeB
+
+```ts
+import { Prototype } from './prototype.interface';
+
+export class ConcretePrototypeB implements Prototype {
+    public stateB: number;
+
+    constructor(stateB: number) {
+        this.stateB = stateB;
+    }
+
+    clone(): ConcretePrototypeB {
+        return new ConcretePrototypeB(this.stateB);
+    }
+}
+```
```

Comment:
Adicionar aqui análise dos resultados. Vi que foi criado apenas um exemplo base de protatype, se não for aplicado ao projeto, justificar o porquê. Acredito que seja, então também mudar o diagrama e os códigos **aplicando** ao projeto. Em qual(is) classes usar etc